From 6b5e0ad992750e81c8eb9118b52179b8fad4426b Mon Sep 17 00:00:00 2001
From: Baruch Siach <baruch@tkos.co.il>
Date: Sun, 12 Mar 2017 19:46:52 +0200
Subject: [PATCH] btrfs-progs: add missing limits headers

Build under musl libc fails because of missing PATH_MAX and XATTR_NAME_MAX
macro declarations. Add the required headers.

Signed-off-by: Baruch Siach <baruch@tkos.co.il>
---
Upstream status: https://github.com/kdave/btrfs-progs/pull/28

 convert/source-ext2.c | 1 +
 mkfs/common.c         | 1 +
 2 files changed, 2 insertions(+)

diff --git a/convert/source-ext2.c b/convert/source-ext2.c
index 824d45d063d4..77ce0c66ac2a 100644
--- a/convert/source-ext2.c
+++ b/convert/source-ext2.c
@@ -16,6 +16,7 @@
 
 #if BTRFSCONVERT_EXT2
 
+#include <linux/limits.h>
 #include "kerncompat.h"
 #include "disk-io.h"
 #include "transaction.h"
diff --git a/mkfs/common.c b/mkfs/common.c
index 1e6a32419b68..1e8f26ea4dda 100644
--- a/mkfs/common.c
+++ b/mkfs/common.c
@@ -18,6 +18,7 @@
 #include <uuid/uuid.h>
 #include <blkid/blkid.h>
 #include <fcntl.h>
+#include <limits.h>
 #include "ctree.h"
 #include "disk-io.h"
 #include "volumes.h"
-- 
2.11.0

